[Bug]: NullPointerException: The mapper [com.appsmith.server.solutions.ce.PluginScheduledTaskCEImpl$$Lambda$2190/0x000000080178e448] returned a null value.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I'm executing `docker-compose up` and expect for the app to run, however, I got the following  error:
```
reactor.core.Exceptions$ErrorCallbackNotImplemented:
java.lang.NullPointerException: The mapper [com.appsmith.server.solutions.ce.PluginScheduledTaskCEImpl$$Lambda$2059/0x00000008016c1748] returned a null value.

Caused by: java.lang.NullPointerException:
The mapper [com.appsmith.server.solutions.ce.PluginScheduledTaskCEImpl$$Lambda$2059/0x00000008016c1748] returned a null value
```

### Steps To Reproduce

1. `curl -L https://bit.ly/3h08WqB -o $PWD/docker-compose.yml`
2. `docker-compose up`

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Self Hosted - latest (v1.9.4)

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ResponseDTO.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ResponseDTO.java
@@ -6,6 +6,7 @@ import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 import lombok.ToString;
+import org.apache.commons.lang3.StringUtils;
 
 import java.io.Serializable;
 
@@ -34,6 +35,19 @@ public class ResponseDTO<T> implements Serializable {
 
     public ResponseDTO(int status, ErrorDTO errorDTO) {
         this.responseMeta = new ResponseMetaDTO(status, errorDTO);
+    }
+
+    public String getErrorDisplay() {
+        if (responseMeta == null) {
+            return "";
+        }
+
+        final ErrorDTO error = responseMeta.getError();
+        if (error == null || error.getMessage() == null) {
+            return "";
+        }
+
+        return error.getCode() + ": " + error.getMessage();
     }
 
 }

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/PluginScheduledTaskCEImpl.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/PluginScheduledTaskCEImpl.java
@@ -42,7 +42,7 @@ public class PluginScheduledTaskCEImpl implements PluginScheduledTaskCE {
 
 
     // Number of milliseconds between the start of each scheduled calls to this method.
-    @Scheduled(initialDelay = 3 * 1000 /* 3 seconds */, fixedRate = 2 * 60 * 60 * 1000 /* two hours */)
+    @Scheduled(initialDelay = 30 * 1000 /* 30 seconds */, fixedRate = 2 * 60 * 60 * 1000 /* two hours */)
     public void updateRemotePlugins() {
         // Get all plugins on this instance
         final Mono<Map<PluginIdentifier, Plugin>> availablePluginsMono =

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/PluginScheduledTaskCEImpl.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/PluginScheduledTaskCEImpl.java
@@ -4,6 +4,7 @@ import com.appsmith.server.configurations.CloudServicesConfig;
 import com.appsmith.server.domains.Plugin;
 import com.appsmith.server.domains.Workspace;
 import com.appsmith.server.dtos.ResponseDTO;
+import com.appsmith.server.helpers.CollectionUtils;
 import com.appsmith.server.services.ConfigService;
 import com.appsmith.server.services.PluginService;
 import com.appsmith.util.WebClientUtils;
@@ -21,6 +22,7 @@ import reactor.core.scheduler.Schedulers;
 
 import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
@@ -40,7 +42,7 @@ public class PluginScheduledTaskCEImpl implements PluginScheduledTaskCE {
 
 
     // Number of milliseconds between the start of each scheduled calls to this method.
-    @Scheduled(initialDelay = 30 * 1000 /* 30 seconds */, fixedRate = 2 * 60 * 60 * 1000 /* two hours */)
+    @Scheduled(initialDelay = 3 * 1000 /* 30 seconds */, fixedRate = 2 * 60 * 60 * 1000 /* two hours */)
     public void updateRemotePlugins() {
         // Get all plugins on this instance
         final Mono<Map<PluginIdentifier, Plugin>> availablePluginsMono =
@@ -111,7 +113,13 @@ public class PluginScheduledTaskCEImpl implements PluginScheduledTaskCE {
                         .get()
                         .exchangeToMono(clientResponse -> clientResponse.bodyToMono(new ParameterizedTypeReference<ResponseDTO<List<Plugin>>>() {
                         }))
-                        .map(ResponseDTO::getData)
+                        .map(listResponseDTO -> {
+                            if (listResponseDTO.getData() == null) {
+                                log.error("Error fetching plugins from cloud-services. Error: {}", listResponseDTO.getErrorDisplay());
+                                return Collections.<Plugin>emptyList();
+                            }
+                            return listResponseDTO.getData();
+                        })
                         .map(plugins -> {
                             // Set new updated time
                             this.lastUpdatedAt = Instant.now();

--- a/app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/PluginScheduledTaskCEImpl.java
+++ b/app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/PluginScheduledTaskCEImpl.java
@@ -42,7 +42,7 @@ public class PluginScheduledTaskCEImpl implements PluginScheduledTaskCE {
 
 
     // Number of milliseconds between the start of each scheduled calls to this method.
-    @Scheduled(initialDelay = 3 * 1000 /* 30 seconds */, fixedRate = 2 * 60 * 60 * 1000 /* two hours */)
+    @Scheduled(initialDelay = 3 * 1000 /* 3 seconds */, fixedRate = 2 * 60 * 60 * 1000 /* two hours */)
     public void updateRemotePlugins() {
         // Get all plugins on this instance
         final Mono<Map<PluginIdentifier, Plugin>> availablePluginsMono =

